src/x509/cert.ts

```diff
@@ -157,14 +162,71 @@ export class x509Certificate {
     return this.root.raw.equals(other.root.raw);
   }
 
-  private findExtension(oid: string): ASN1Obj | undefined {
-    // The extension list is the first (and only) element of the extensions
-    // context specific tag
-    const extSeq = this.extensionsObj?.subs[0];
+  public verifySCTs(
+    issuer: x509Certificate,
+    logs: sigstore.TransparencyLogInstance[]
+  ): boolean {
+    let extSCT: x509SCTExtension | undefined;
+
+    // Verifying the SCT requires that we remove the SCT extension and
+    // re-encode the TBS structure to DER -- this value is part of the data
+    // over which the signature is calculated. Since this is a destructive action
+    // we create a copy of the certificate so we can remove the SCT extension
+    // without affecting the original certificate.
+    const clone = this.clone();
+
+    // Intentionally not using the findExtension method here because we want to
+    // remove the the SCT extension from the certificate before calculating the
+    // PreCert structure
+    for (let i = 0; i < clone.extensions.length; i++) {
+      const ext = clone.extensions[i];
+
+      if (ext.subs[0].toOID() === EXTENSION_OID_SCT) {
+        extSCT = new x509SCTExtension(ext);
+
+        // Remove the extension from the certificate
+        clone.extensions.splice(i, 1);
+        break;
+      }
+    }
 
+    if (!extSCT) {
+      throw new Error('Certificate does not contain SCT extension');
+    }
+
+    if (extSCT?.signedCertificateTimestamps?.length === 0) {
+      throw new Error('Certificate does not contain any SCTs');
+    }
+
+    // Construct the PreCert structure
+    // https://www.rfc-editor.org/rfc/rfc6962#section-3.2
+    const preCert = new ByteStream();
+
+    // Calculate hash of the issuer's public key
+    const issuerId = crypto.hash(issuer.publicKey);
+    preCert.appendView(issuerId);
+
+    // Re-encodes the certificate to DER after removing the SCT extension
+    const tbs = clone.tbsCertificate.toDER();
+    preCert.appendUint24(tbs.length);
+    preCert.appendView(tbs);
+
+    return extSCT.signedCertificateTimestamps.every((sct) =>
+      sct.verify(preCert.buffer, logs)
+    );
```

Comment:
When I integrate this into the top-level verification logic, I'll likely refactor this to return an array of results (or maybe the count of the number of verified SCTs) so that we can compare to the verification threshold.